.scalafmt.conf

```diff
@@ -0,0 +1,3 @@
+version=2.5.2
+
+maxColumn = 100
```

Comment:
@nicl I think the settings are not the best for readability.
Below I am sharing few examples noticed where I think readability degrades

Before
```scala
 lazy val dfpFacebookIaAdUnitCacheJob: DfpFacebookIaAdUnitCacheJob = wire[DfpFacebookIaAdUnitCacheJob]
```

After
```scala
   lazy val dfpFacebookIaAdUnitCacheJob: DfpFacebookIaAdUnitCacheJob =
     wire[DfpFacebookIaAdUnitCacheJob]
```

Before:

```scala
  getMap(articlePage, Edition(request), false).getOrElse("hasShowcaseMainElement", JsBoolean(false)).as[Boolean],
       getMap(articlePage, Edition(request), false).getOrElse("isFront", JsBoolean(false)).as[Boolean],
       getMap(articlePage, Edition(request), false).getOrElse("isLiveBlog", JsBoolean(false)).as[Boolean],
       getMap(articlePage, Edition(request), false).getOrElse("isMinuteArticle", JsBoolean(false)).as[Boolean],
       getMap(articlePage, Edition(request), false).getOrElse("isPaidContent", JsBoolean(false)).as[Boolean],
```

After 
```scala
       getMap(articlePage, Edition(request), false)
         .getOrElse("hasShowcaseMainElement", JsBoolean(false))
         .as[Boolean],
       getMap(articlePage, Edition(request), false)
         .getOrElse("isFront", JsBoolean(false))
         .as[Boolean],
       getMap(articlePage, Edition(request), false)
         .getOrElse("isLiveBlog", JsBoolean(false))
         .as[Boolean],
       getMap(articlePage, Edition(request), false)
         .getOrElse("isMinuteArticle", JsBoolean(false))
         .as[Boolean],
       getMap(articlePage, Edition(request), false)
         .getOrElse("isPaidContent", JsBoolean(false))
         .as[Boolean],
```

Before
```scala
 def comments(key: DiscussionKey): Action[AnyContent] = Action.async { implicit request => getComments(key) }
 def commentsJson(key: DiscussionKey): Action[AnyContent] = Action.async { implicit request => getComments(key) }
 def commentsJsonOptions(key: DiscussionKey): Action[AnyContent] = Action { implicit request => TinyResponse.noContent(Some("GET, OPTIONS")) }
```

After
```scala
   def comments(key: DiscussionKey): Action[AnyContent] =
     Action.async { implicit request => getComments(key) }
   def commentsJson(key: DiscussionKey): Action[AnyContent] =
     Action.async { implicit request => getComments(key) }
   def commentsJsonOptions(key: DiscussionKey): Action[AnyContent] =
     Action { implicit request => TinyResponse.noContent(Some("GET, OPTIONS")) }
```

Reply:
Thanks @mchv . It's tough because this is obviously so subjective and people are bound to disagree. For what it's worth, I prefer the 'after' versions in your examples!

There's a good list of line length recommendations across languages here:

https://en.wikipedia.org/wiki/Characters_per_line

The majority of which fall below 100. The Scala style guide itself doesn't recommend anything clearly here unfortunately, but does allude to anything over 80 as being sensible for wrapping:

https://docs.scala-lang.org/style/indentation.html#line-wrapping

Reply:
Just want to say that I *much* prefer the before version.

Reply:
For those who would prefer a greater characters per line setting - cc @mchv @shtukas - what would you prefer?

The other thing I'd add is that once you go over about 100 you stop being able to get two code windows open side by side on a standard resolution (or simply, if you have one coding window and then say some docs etc on the other half), which is another thing to consider.

Reply:
fwiw, prettier default line length is 80 chars

Reply:
I agree that 100 is not the best use of available space for coding in an IDE on a decent monitor, but I'm also conscious that we're only 2 steps away from spaces vs tabs debate! 😓 However I wanted to point out we need to consider more than one case here.

If you're reading diffs in github or merging branches in intellij, clarity is maintained better with short lines, especially with side by side diff.  Ironically in @mchv comment the shorter lines are more readable because of no need to scroll sideways
![image](https://user-images.githubusercontent.com/7304387/91159689-4d1bd380-e6c0-11ea-91f9-f309bf920de3.png)

Also we are discussing the current code being auto reformatted which is never going to be perfect.  Having a shorter line length will encourage people to write their code differently to fit in with that.

At the end of the day everyone is going to have a different figure based of how they read code more often, my personal opinion is to allow more than 100 but encourage short lines where possible.  Code is communication/documentation so I think we should take more account of what people who use the codebase think than recommendations online, because some of them will still be based around the 80 character standard terminal size!

Reply:
Cc @mchv @shtukas @tomrf1 and @johnduffell . Thanks for feedback on this. I've set this to 120 for now, which is used in a few Scala projects and 'fixes' the examples above. Depending on how it goes, we might consider reducing this further in future, but 120 feels like a good compromise at the moment that isn't too restrictive but still provides some kind of upper bound.

Reply:
I find these to be far better readability, but that's likely because I am used to look at every JS / TS codebase with prettier on. I guess that's the point - when you're used to reading code with a standard formatting style, there's a far lower cognitive load when coming to an unfamiliar codebase (or unfamiliar part of a huge codebase!).